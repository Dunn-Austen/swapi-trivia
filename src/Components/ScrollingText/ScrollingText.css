 section {
  background: 100%;
  height: 100%;
  overflow: hidden;
  width: 100%;
}

.scrolling-container {
  background: black;
  color: yellow;
  display: flex;
  justify-content: center;
  height: 800px;
  perspective: 400px;
  position: relative;
  text-align: justify;

}

.scrolling-content {
  animation: scrollAnimation 50s linear;
  /* Reference comments included from online guide to 'Crawl component' */
  /* Reference URL - https://css-tricks.com/snippets/css/star-wars-crawl-text/ */
  position: relative;
  top: -100px;
  transform-origin: 50% 100%;
}

@keyframes scrollAnimation {
  0% {
    top: 0;
    transform: rotateX(20deg) translateZ(0);
  }
  100% {
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}
