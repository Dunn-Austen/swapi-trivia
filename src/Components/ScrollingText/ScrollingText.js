import React from 'react';
import './ScrollingText.css';

const ScrollingText = () => {

  return (
    <section className="scrolling-text">
      <p>Placeholder Text, need to research how to get the animation for the text working</p>
    </section>
  )
};

export default ScrollingText;
