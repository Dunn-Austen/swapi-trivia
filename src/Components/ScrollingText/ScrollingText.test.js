import React from 'react';
import { shallow } from 'enzyme';
import ScrollingText from '../ScrollingText/ScrollingText';

describe('ScrollingText', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<ScrollingText
      crawl={'text'}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
