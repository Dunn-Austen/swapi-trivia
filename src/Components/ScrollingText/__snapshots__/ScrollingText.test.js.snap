// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollingText should match the snapshot with all data passed in correctly 1`] = `
<section
  className="scroll-section"
>
  <div
    className="scrolling-container"
  >
    <div
      className="scrolling-content"
    >
      <p>
        text
      </p>
    </div>
  </div>
</section>
`;
